lua/nvim_lsp/zls.lua

```diff
@@ -0,0 +1,20 @@
+local configs = require 'nvim_lsp/configs'
+local util = require 'nvim_lsp/util'
+
+configs.zls = {
+    default_config = {
+        cmd = {"zls"};
+        filetypes = {"zig", "zir"};
+        root_dir = util.root_pattern("zls.json", ".git", ".");
```

Comment:
Does using `.` work correctly? I think most other configs are using something like this where `util.path.dirname` returns the path to the directory of the opened file. But using your approach definitely looks appealing.

```lua
root_dir = function(fname)
  return util.root_pattern("zls.json", ".git")(fname) or util.path.dirname(fname)
end
```

Reply:
I thinks it's works correctly, I used this config for some days, and I think using `"."` is better than `util.path.dirname` because, with this approch you just need to cd to the root of your project and everything will work smoothly, with `util.path.dirname` some autocompletion will not appear ( e.g. files in subdirectory, etc ).

Sorry for my English, it's not my native language

Reply:
Edit:
using `util.path.dirname` is better, sometime with `"."`, when creating a new file autocompletion would stop working